### PR TITLE
Prompt Component

### PR DESCRIPTION
**Pull request by ElliotTheHuman**
### 🔧 To-Do
- Add documentation on `Prompt`
- Figure out why `IconButton` test gets borked

### :sparkles: Changes
- Added Prompt component
- Added Prompt tests

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [ ] Build and tests are passing
- [x] PR is ideally < 400LOC

### :camera: Screenshots
prompt

### CODE REVIEW

File: packages/components/src/Modal/Dialog/PromptDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  /**
+   * Defines the color of the cancel button. Can be the string name of a color listed in the color theme, or a color object.
+   * @default "neutral"
+   */
+  cancelColor?: keyof SemanticColors
+  /**
+   * Defines the color of the save button. Can be the string name of a color listed in the color theme, or a color object.
+   * @default "primary"
+   */
+  buttonColor?: keyof SemanticColors
```
Comment (lukelooker) [inline packages/components/src/Modal/Dialog/PromptDialog.tsx:48]: I think we should skip this for now. I don't see a case where a Prompt needs a different button color where-as Confirm is often used for a Delete confirmation.

```suggestion
```

Comment (ElliotTheHuman) [reply packages/components/src/Modal/Dialog/PromptDialog.tsx]: Addressed in https://github.com/looker-open-source/components/pull/757/commits/b6a172277fdd866ddb7fadfb99e169ad3f77febd!

File: packages/components/src/Modal/Prompt/PromptDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+//             {saveLabel}
+//           </Button>
+//           <ButtonTransparent type="reset" color={cancelColor} onClick={cancel}>
+//             {cancelLabel}
+//           </ButtonTransparent>
+//         </ModalFooter>
+//       </form>
+//     </Dialog>
+//   )
+// }
```
Comment (lukelooker) [inline packages/components/src/Modal/Prompt/PromptDialog.tsx:276]: Remove unused code.

```suggestion
```

File: packages/components/src/Modal/Prompt/index.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+export * from './Prompt'
```
Comment (lukelooker) [inline packages/components/src/Modal/Prompt/index.tsx:27]: This should be named `index.ts` (not `.tsx`)

File: packages/components/src/Modal/index.ts [hunk 4]
```diff
@@ -24,9 +24,11 @@
 
  */
 
+export * from './Confirm'
 export * from './Dialog'
 export * from './Drawer'
 export * from './Layout'
+export * from './Prompt'
 
 export * from './Modal'
```
Comment (lukelooker) [inline packages/components/src/Modal/index.ts:33]: Random whitespace here... once it's removed you'll need to re-order slightly.

File: packages/components/src/Modal/Prompt/index.tsx (hunk 3 above)
Comment (ElliotTheHuman) [reply packages/components/src/Modal/Prompt/index.tsx]: Resolved in 0215229!

File: packages/components/src/Modal/index.ts (hunk 4 above)
Comment (ElliotTheHuman) [reply packages/components/src/Modal/index.ts]: Resolved in 0215229!

**Pull request merged**